### PR TITLE
[server] remove /tos endpoint – WEB-121

### PR DESCRIPTION
**Pull request by AlexTugarev**
This removes the obsolete ToS parts from server, while trying not to touch the current behavior. 

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-rm-tos</li>
	<li><b>🔗 URL</b> - <a href="https://at-rm-tos.preview.gitpod-dev.com/workspaces" target="_blank">at-rm-tos.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
</ul>

## Related Issue(s)

Part of WEB-121

## How to test
1. Check that sign-in, log-out, and log-in does work as expected. 
2. Check that Git Integrations do work, e.g. updating scopes.

## Release Notes

```release-note
NONE
```

## Build Options:

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`

<details>
<summary>Publish Options</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer Options</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] with-ws-manager-mk2
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

#### Preview Environment Options:
- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

/hold

### CODE REVIEW

File: components/server/src/user/user-service.ts [hunk 1]
```diff
@@ -425,41 +384,6 @@ export class UserService {
         await this.userDb.storeSingleToken(candidate, token);
     }
 
-    async updateUserEnvVarsOnLogin(user: User, envVars?: UserEnvVarValue[]) {
```
Comment (easyCZ) [inline components/server/src/user/user-service.ts:428]: Is this no longer used? It doesn't seem to be related to the TOS flow

File: components/server/src/auth/generic-auth-provider.ts [hunk 2]
```diff
@@ -809,7 +812,35 @@ export class GenericAuthProvider implements AuthProvider {
     };
 }
 
-export type VerifyResult = TosFlow.WithIdentity | TosFlow.WithUser;
+interface VerifyResult {
```
Comment (AlexTugarev) [inline components/server/src/auth/generic-auth-provider.ts:815]: The shape of `TosFlow` was used as result, thus moving it here, while getting rid of everything from session's storage. This is now basically the return type of the verify function in the auth provider.

File: components/server/src/user/user-service.ts (hunk 1 above)
Comment (AlexTugarev) [reply components/server/src/user/user-service.ts:428]: ToS flow payload used to capture these. It didn't make sense to move the call site of this when moving the remainings of the `TosFlow` shape, which is now a plain result type of a function and doesn't represent a state of the session. Short: it's dead code. There is no use case for deriving/setting env vars on logins, which was a SH special anyways.

File: components/server/src/auth/generic-auth-provider.ts [hunk 3]
```diff
@@ -446,25 +449,29 @@ export class GenericAuthProvider implements AuthProvider {
...
-                log.info(context, `(${strategyName}) Redirect to /api/tos`, logPayload);
-
-                // attach the sign up info to the session, in order to proceed after acceptance of terms
-                await TosFlow.attach(request.session!, flowContext);
-
-                response.redirect(this.config.hostUrl.withApi({ pathname: "/tos", search: "mode=login" }).toString());
-                return;
+
+            if (VerifyResult.WithIdentity.is(flowContext)) {
+                log.info(context, `(${strategyName}) Creating new user and complete login.`, logPayload);
```
Comment (AlexTugarev) [inline components/server/src/auth/generic-auth-provider.ts:458]: – To reviewers –

This is the significant part of the whole PR. Instead of redirecting to /tos endpoint, here the two steps are inlined which would happen otherwise inside of the old /tos handler:
1. creating new account
2. finishing the login

Comment (AlexTugarev) [reply components/server/src/auth/generic-auth-provider.ts:458]: Previous location of these steps is https://github.com/gitpod-io/gitpod/pull/17236/files#diff-0fda9115576b7e1168858f5d4f8869e3120c3a8364cc8f43335037ebb78f6734L573

(Search for `router.get("/tos"`, if the link does not work.)

**Review by easyCZ — APPROVED** [review]

**Comment by AlexTugarev** [comment]
I'll gonna retest on staging once more.

File: components/server/src/user/user-service.ts (hunk 1 above)
Comment (randomir) [reply components/server/src/user/user-service.ts:436]: Hey @AlexTugarev! While that may as well be the case for gitpod.io, we actually do need to set an environment variable or two for some users, in some environments (installs), e.g. in a test env.

With this code removed, could you suggest/recommend a flow or a place in code where would be the best to set user env vars? Ideally a place that's relatively safe/robust going forward (won't be yanked anytime soon). Thanks!

/cc @charleswhchan
